tests/end_to_end/end_to_end.cpp

```diff
@@ -673,3 +680,106 @@ TEST_CASE("pinned_map_enum", "[end_to_end]")
 
     ebpf_test_pinned_map_enum();
 }
+
+TEST_CASE("implicit_explicit_detach", "[end_to_end]")
+{
+    _test_helper_end_to_end test_helper;
+
+    ebpf_handle_t program_handle;
+    ebpf_handle_t map_handle;
+    uint32_t count_of_map_handle = 1;
+    uint32_t result = 0;
+    const char* error_message = NULL;
+    ebpf_handle_t link_handle;
+
+    single_instance_hook_t hook(EBPF_PROGRAM_TYPE_XDP);
+    program_info_provider_t xdp_program_info(EBPF_PROGRAM_TYPE_XDP);
+
+    REQUIRE(
```

Comment:
You could move the subcases as different test functions. Or at least put a comment above every sub-case stating what it tests.

Reply:
Done both.